app/narwos.py

```diff
@@ -273,6 +277,52 @@ def show_clusters():
     
     return 'Nothing to show here'
 
+@app.route('/show_filters_submit', methods=['POST'])
+def show_filter_submit():
+
+    show_filter = request.form.get('show_filter')
+    if show_filter:
+        return redirect(url_for('show_filter', show_filter=show_filter))
+    else:
+        return redirect(url_for("index", message=f"Cannot show the filtering!"))
+
+@app.route('/show_filters', methods=['GET'])
+def show_filter():
+
+    df = pd.read_csv(PATH_TO_CURRENT_DF)
+
+    if isinstance(df, pd.DataFrame):
+
+        return render_template('filtering.html', columns=df.columns)
+    
+    return 'Nothing to show here'
+
+@app.route('/filter', methods=['POST'])
+def filter_data():
+    filters = request.get_json()
+    filtered_df = pd.read_csv(PATH_TO_CURRENT_DF)
```

Comment:
the same as above and why do you have to read df twice? Is it not possible to keep it memory? But do not use flask.session object becuase object is to big, but you can try to experiment with g object and serialization, maybe you can find a better aproach then reading file twice